### PR TITLE
fix: ensure BAZEL_NODE_RUNFILES_HELPER & BAZEL_NODE_PATCH_REQUIRE are absolute

### PR DESCRIPTION
**Pull request by gregmagolan**
This is for using `BAZEL_NODE_RUNFILES_HELPER ` in `bazel build` when there are no runfiles and with a tool that does not support workers. The cwd is a path such as: `/private/var/tmp/_bazel_greg/35306ad70b8da8f7bc31590523de35c0/sandbox/darwin-sandbox/18/execroot/build_bazel_rules_nodejs`.

```
export BAZEL_NODE_RUNFILES_HELPER=$(rlocation "build_bazel_rules_nodejs/internal/linker/runfiles_helper.js")
```

resolves to `/Users/greg/google/rules_nodejs/internal/linker/runfiles_helper.js` when workers are on as there is no sandbox (such as ts_library).

&

resolves to `bazel-out/host/bin/external/build_bazel_rules_typescript/internal/tsc_wrapped_bin.sh.runfiles/build_bazel_rules_nodejs/internal/linker/runfiles_helper.js` when workers are off as there is a sandbox

The latter is a relative path to the CWD and won't work in a require() statement unless joined with the CWD.

This PR adds the CWD in the launcher.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature (please, look at the "Scope of the project" section in the README.md file)
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: internal/node/launcher.sh [hunk 1]
```diff
@@ -142,10 +142,16 @@ fi
 
 # Export the location of the runfiles helpers script
 export BAZEL_NODE_RUNFILES_HELPER=$(rlocation "TEMPLATED_runfiles_helper_script")
+if [[ "${BAZEL_NODE_RUNFILES_HELPER}" != /* ]]; then
```
Comment (Toxicable) [inline internal/node/launcher.sh]: Does this account for windows?

Comment (gregmagolan) [reply internal/node/launcher.sh]: I believe so as the this in an .sh script so even Windows absolute paths start with `/C/path/to/file`. Will fire up my Windows machine to be sure however.

Comment (Toxicable) [reply internal/node/launcher.sh]: Oh right, I didn't know that, that's super nice if your'e right

Comment (gregmagolan) [reply internal/node/launcher.sh]: Looks like you were right after all. $(rlocation) returns `C:/path/to/file` for absolute paths on Windows.

Comment (Toxicable) [reply internal/node/launcher.sh]: ah shoot, oh well, good to know!

File: internal/node/test/test_runfiles_helper.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+const fs = require('fs');
+const args = process.argv.slice(2);
+const outfile = args.shift();
+const runfiles = require(process.env['BAZEL_NODE_RUNFILES_HELPER']);
+const golden =
```
Comment (alexeagle) [inline internal/node/test/test_runfiles_helper.js]: this line looks strange, lots of spaces for some reason?

Comment (gregmagolan) [reply internal/node/test/test_runfiles_helper.js]: clang format did that for line length I think

**Pull request merged**